services/event_bus/event_handler.go

```diff
@@ -121,6 +133,16 @@ func handleHAPUpdate(conf *configuration.Configuration, conn *zk.Conn) bool {
 
 	templateData := haproxy.GetTemplateData(conf, conn)
 
+  // Any empty updates from Marathon will not result in any Haproxy updates.
+	// Haproxy will continue to use previous state.
+  if templateData.Apps == nil || len(templateData.Apps) == 0 {
+		isConfigStale = true
+		log.Println("Got no Apps in template data. Skipped haproxy update")
+		return false
+	} else {
```

Comment:
The `else` is not required because you're returning in the if block.
Also, not being stale is not guaranteed yet at this point.

@amitbose327 @vmahedia